features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/BaseCheckableTextViewMapperTest.kt

```diff
@@ -0,0 +1,287 @@
+
+    @Mock
+    lateinit var mockNotCheckedDrawable: Drawable
+
+    @Mock
+    lateinit var mockClonedDrawable: Drawable
+
+    @IntForgery
+    var mockCloneDrawableIntrinsicHeight: Int = 0
+
+    @IntForgery
+    var mockCloneDrawableIntrinsicWidth: Int = 0
+
+    @BeforeEach
+    fun `set up`() {
+        mockClonedDrawable = mock {
+            whenever(it.intrinsicHeight) doReturn mockCloneDrawableIntrinsicHeight
+            whenever(it.intrinsicWidth) doReturn mockCloneDrawableIntrinsicWidth
+        }
+        mockCheckedConstantState = mock {
+            whenever(it.newDrawable(anyVararg(Resources::class))) doReturn mockClonedDrawable
+        }
+        mockCheckedDrawable = mock {
+            whenever(it.constantState) doReturn mockCheckedConstantState
+        }
+        mockNotCheckedDrawable = mock {
+            whenever(it.constantState) doReturn mockCheckedConstantState
+        }
+        mockConstantState = mock {
+            whenever(it.getChild(CHECK_BOX_CHECKED_DRAWABLE_INDEX)) doReturn mockCheckedDrawable
+            whenever(it.getChild(CHECK_BOX_NOT_CHECKED_DRAWABLE_INDEX)) doReturn mockNotCheckedDrawable
+        }
+        mockButtonDrawable = mock {
+            whenever(it.constantState) doReturn mockConstantState
+        }
+
+        mockCheckableTextView = mockCheckableTextView()
+        whenever(
+            mockViewIdentifierResolver.resolveChildUniqueIdentifier(
+                mockCheckableTextView,
+                CheckableTextViewMapper.CHECKABLE_KEY_NAME
+            )
+        ).thenReturn(fakeGeneratedIdentifier)
+
+        whenever(
+            mockTextWireframeMapper.map(
+                eq(mockCheckableTextView),
+                eq(fakeMappingContext),
+                any(),
+                eq(mockInternalLogger)
+            )
+        )
+            .thenReturn(fakeTextWireframes)
+
+        whenever(
+            mockViewBoundsResolver.resolveViewGlobalBounds(
+                mockCheckableTextView,
+                fakeMappingContext.systemInformation.screenDensity
+            )
+        ).thenReturn(fakeViewGlobalBounds)
+
+        whenever(
+            mockColorStringFormatter.formatColorAndAlphaAsHexString(fakeCurrentTextColor, OPAQUE_ALPHA_VALUE)
+        ) doReturn fakeCurrentTextColorString
+
+        testedCheckableTextViewMapper = setupTestedMapper()
+    }
+
+    internal abstract fun setupTestedMapper(): CheckableTextViewMapper<T>
+
+    internal abstract fun mockCheckableTextView(): T
+
+    internal open fun expectedCheckedShapeStyle(checkBoxColor: String): MobileSegment.ShapeStyle? {
+        return if (fakeMappingContext.privacy == SessionReplayPrivacy.ALLOW) {
+            MobileSegment.ShapeStyle(
+                backgroundColor = checkBoxColor,
+                opacity = mockCheckableTextView.alpha
+            )
+        } else {
+            null
+        }
+    }
+
+    // region Unit Tests
+
+    @TestTargetApi(Build.VERSION_CODES.M)
```

Comment:
Beware about the pitfalls of `TestTargetApi` usage: it will set `Build.VERSION.SDK_INT`, but since it is `static final` field, at some point JIT may inline it, and reflection changes won't have any effect (usually happens after few dozens of calls to this field).

Reply:
Good to know, if there is any modification that I should apply on this test?

Reply:
ideally we could inject `BuildSdkVersionProvider` wrapper which we can read an mock, but we can leave it for the later refactoring.